examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.cpp

```diff
@@ -632,115 +627,84 @@ static void wfx_rsi_save_ap_info() // translation
  **********************************************************************************************/
 static sl_status_t wfx_rsi_do_join(void)
 {
+    ReturnErrorCodeIf((wfx_rsi.dev_state & (WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED)), SL_STATUS_IN_PROGRESS);
     sl_status_t status = SL_STATUS_OK;
-    sl_wifi_security_t connect_security_mode;
+    sl_wifi_client_configuration_t ap;
+    memset(&ap, 0, sizeof(sl_wifi_client_configuration_t));
     WfxEvent_t event;
     switch (wfx_rsi.sec.security)
     {
     case WFX_SEC_WEP:
-        connect_security_mode = SL_WIFI_WEP;
+        ap.security = SL_WIFI_WEP;
         break;
     case WFX_SEC_WPA:
-        connect_security_mode = SL_WIFI_WPA_WPA2_MIXED;
+        ap.security = SL_WIFI_WPA_WPA2_MIXED;
         break;
     case WFX_SEC_WPA2:
 #if WIFI_ENABLE_SECURITY_WPA3_TRANSITION
-        connect_security_mode = SL_WIFI_WPA3_TRANSITION;
+        ap.security = SL_WIFI_WPA3_TRANSITION;
         break;
     case WFX_SEC_WPA3:
-        connect_security_mode = SL_WIFI_WPA3_TRANSITION;
+        ap.security = SL_WIFI_WPA3_TRANSITION;
 #else
-        connect_security_mode = SL_WIFI_WPA_WPA2_MIXED;
+        ap.security          = SL_WIFI_WPA_WPA2_MIXED;
 #endif // WIFI_ENABLE_SECURITY_WPA3_TRANSITION
         break;
     case WFX_SEC_NONE:
-        connect_security_mode = SL_WIFI_OPEN;
+        ap.security = SL_WIFI_OPEN;
         break;
     default:
-        SILABS_LOG("error: unknown security type.");
+        ChipLogError(DeviceLayer, "wfx_rsi_do_join: unknown security type.");
         return status;
     }
 
-    if (wfx_rsi.dev_state & (WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED))
-    {
-        SILABS_LOG("%s: not joining - already in progress", __func__);
-    }
-    else
-    {
-        SILABS_LOG("%s: WLAN: connecting to %s, sec=%d", __func__, &wfx_rsi.sec.ssid[0], wfx_rsi.sec.security);
-
-        /*
-         * Join the network
-         */
-        /* TODO - make the WFX_SECURITY_xxx - same as RSI_xxx
-         * Right now it's done by hand - we need something better
-         */
-        wfx_rsi.dev_state |= WFX_RSI_ST_STA_CONNECTING;
+    ChipLogDetail(DeviceLayer, "wfx_rsi_do_join: SSID: %s, SECURITY: %d(%d)", &wfx_rsi.sec.ssid[0], ap.security,
```

Comment:
Logging of SSID is done prior to validating it is nul-terminated. If it is assumed null-terminated, then have an `if` that verifies that it is null-terminated.